client/my-sites/domains/domain-management/list/bulk-edit-contact-info.jsx

```diff
@@ -0,0 +1,219 @@
+/**
+ * External dependencies
+ */
+
+import PropTypes from 'prop-types';
+import { localize } from 'i18n-calypso';
+import React from 'react';
+import { connect } from 'react-redux';
+import { Button } from '@automattic/components';
+
+/**
+ * Internal dependencies
+ */
+import DomainContactDetails from 'calypso/my-sites/checkout/composite-checkout/components/domain-contact-details';
+import { fetchUserSettings } from 'calypso/state/user-settings/actions';
+import { recordTracksEvent } from 'calypso/state/analytics/actions';
+import getContactDetailsCache from 'calypso/state/selectors/get-contact-details-cache';
+import isRequestingContactDetailsCache from 'calypso/state/selectors/is-requesting-contact-details-cache';
+import { requestContactDetailsCache, saveWhois } from 'calypso/state/domains/management/actions';
+import getUserSettings from 'calypso/state/selectors/get-user-settings';
+import { errorNotice, infoNotice, successNotice } from 'calypso/state/notices/actions';
+import { isFetchingUserSettings } from 'calypso/state/user-settings/selectors';
+import TransferLockOptOut from 'calypso/my-sites/domains/domain-management/list/transfer-lock-opt-out';
+import wp from 'calypso/lib/wp';
+
+const wpcom = wp.undocumented();
+
+/**
+ * Style dependencies
+ */
+import './list-all.scss';
+
+class BulkEditContactInfo extends React.Component {
+	static propTypes = {
+		handleSaveContactInfo: PropTypes.func.isRequired,
+		isDisabled: PropTypes.bool,
+		isSubmitting: PropTypes.bool,
+		onTransferLockOptOutChange: PropTypes.func.isRequired,
+		emailOnly: PropTypes.bool,
+		domainNamesList: PropTypes.array,
+	};
+
+	static defaultProps = {
+		isDisabled: false,
+		isSubmitting: false,
+		emailOnly: false,
+	};
+
+	state = {
+		hasSetContactDetailsFromCache: false,
+		hasSetInitialContactEmail: false,
+		contactDetails: {},
+		errorMessages: {},
+	};
+
+	componentDidMount() {
+		if ( Object.keys( this.props.userSettings ?? {} ).length === 0 ) {
+			this.props.fetchUserSettings();
+		}
+
+		if ( Object.keys( this.props.contactDetailsCache ?? {} ).length === 0 ) {
+			this.props.requestContactDetailsCache();
+		}
+
+		this.debouncedValidateContactDetails = this.debounce( this.validateContactDetails, 500 );
+	}
+
+	componentDidUpdate() {
+		const { new_user_email, user_email } = this.props.userSettings ?? {};
+		const email = new_user_email?.length > 0 ? new_user_email : user_email;
+
+		if (
+			( email ?? '' ).length > 0 &&
+			( this.state.contactDetails?.email ?? '' ).length === 0 &&
+			false === this.state.hasSetInitialContactEmail
+		) {
+			this.setInitialContactEmail( email );
+		}
+
+		if (
+			Object.keys( this.props.contactDetailsCache ?? {} ).length > 0 &&
+			false === this.state.hasSetContactDetailsFromCache
+		) {
+			const contactDetailsCopy = JSON.parse( JSON.stringify( this.props.contactDetailsCache ) );
```

Comment:
At first, I didn't understand the reason for this, but then I saw the comment: https://github.com/Automattic/wp-calypso/pull/53481#discussion_r657239076.
`JSON.parse(JSON.stringify())` could be good to create deep copies of an object, but here it looks a bit hacky, and I also don't think a deep copy is necessary... Or is it? 🤔 

The same result could be achieved by doing something like: 

```javascript
const { email, ...contactDetailsCache } = this.props.contactDetailsCache;
this.setContactDetailsFromCache( contactDetailsCache );
```

And this would also remove the `delete` statement on the `setContactDetailsFromCache` method 😸

Reply:
Great suggestion.   Much better.  Thank you!